editor/tree_search.h

```diff
@@ -0,0 +1,126 @@
+/**
+ * tree_search.h
+ * =============================================================================
+ * Copyright 2021-2024 Serhii Snitsaruk
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ * =============================================================================
+ */
+
+#ifdef TOOLS_ENABLED
+
+#ifndef TREE_SEARCH_H
+#define TREE_SEARCH_H
+
+#ifdef LIMBOAI_GDEXTENSION
+#include <godot_cpp/classes/check_box.hpp>
+#include <godot_cpp/classes/h_flow_container.hpp>
+#include <godot_cpp/classes/label.hpp>
+#include <godot_cpp/classes/line_edit.hpp>
+#include <godot_cpp/classes/tree.hpp>
+#include <godot_cpp/templates/hash_map.hpp>
+#endif // LIMBOAI_GDEXTENSION
+
+#ifdef LIMBOAI_MODULE
+#include "scene/gui/check_box.h"
+#include "scene/gui/flow_container.h"
+#include "scene/gui/label.h"
+#include "scene/gui/line_edit.h"
+#include "scene/gui/tree.h"
+#include "core/templates/hash_map.h"
+#endif // LIMBOAI_MODULE
+
+using namespace godot;
+
+enum TreeSearchMode {
```

Comment:
Since we want to eventually separate the TreeSearchPanel from TreeSearch, I propose making TreeSearchMode a member of TreeSearch.

That requires TreeSearch to be predecessor to TreeSearchPanel in the header while forward declaring the TreeSearchPanel. Done.